src/components/DayPicker.jsx

```diff
@@ -753,6 +751,46 @@ class DayPicker extends React.Component {
 
     const isHorizontalAndAnimating = this.isHorizontal() && isCalendarMonthGridAnimating;
 
+    const calendarInfoPositionTop = calendarInfoPosition === INFO_POSITION_TOP;
+    const calendarInfoPositionBottom = calendarInfoPosition === INFO_POSITION_BOTTOM;
+    const calendarInfoPositionLeft = calendarInfoPosition === INFO_POSITION_LEFT;
+    const calendarInfoPositionRight = calendarInfoPosition === INFO_POSITION_RIGHT;
+    const calendarInfoIsInline = calendarInfoPositionLeft || calendarInfoPositionRight;
+
+    const calendarInfo = renderCalendarInfo && (
+      <div
+        ref={this.setCalendarInfoRef}
+        {...css((calendarInfoIsInline) && styles.DayPicker_calendarInfo__horizontal)}
+      >
+        { renderCalendarInfo() }
+      </div>
+    );
+
+    const calendarInfoWith = renderCalendarInfo && calendarInfoIsInline
+      ? calculateDimension(this.calendarInfo, 'width', true, true)
+      : 0;
+
+    const firstVisibleMonthIndex = this.getFirstVisibleIndex();
+    const wrapperHorizontalWidth = (calendarMonthWidth * numberOfMonths) + (2 * DAY_PICKER_PADDING);
+    const fullHorizontalWidth = wrapperHorizontalWidth + calendarInfoWith;
+
+    const transitionContainerStyle = {
+      width: this.isHorizontal() && wrapperHorizontalWidth,
+      height,
+    };
+
+    const dayPickerWrapperStyle = {
+      width: wrapperHorizontalWidth,
+    };
+
+    const dayPickerStyle = {
+      width: this.isHorizontal() && fullHorizontalWidth,
```

Comment:
can we cache the `isHorizontal` result in a var, so we don't have to call the function multiple times?

Reply:
Done.. but the function is called at other different places in the code... Will it be better to cache the result in a state?

Reply:
nah, that would cause a rerender. It's fine to call multiple times conceptually, but it's nice to minimize duplicate function calls within the body of a single function :-)

Reply:
So, it is okay like that ?